spark-3.1-spanner-lib/src/main/java/com/google/cloud/spark/spanner/SpannerUtils.java

```diff
@@ -34,10 +38,24 @@
 import org.apache.spark.sql.catalyst.InternalRow;
 import org.apache.spark.sql.catalyst.expressions.GenericInternalRow;
 import org.apache.spark.sql.types.Decimal;
+import org.apache.spark.unsafe.types.UTF8String;
 
 public class SpannerUtils {
+  public static Map<String, String> defaultConnOpts =
+      new HashMap<String, String>() {
```

Comment:
Please remove explicit project id, instance id, table id etc. from the code and read them from the environment or generate for tests

Reply:
Yup, all removed!